### PR TITLE
build-scripts: stop using do_build.sh

### PR DESCRIPTION
**Pull request by jean-edouard**
- Use git submodules instead of setupoe (and fix do_build.sh so the old way still works)
- Call bb directly from oe/build.sh to build and copy images
- Build the repository, netboot, isos, ... directly in build.sh
- Sorry for the big pull request, I can split it if needed

OXT-507

Signed-off-by: Jed <lejosnej@ainfosec.com>

### CODE REVIEW

**Comment by jean-edouard** [comment]
Note: The builders use a modified version of the build scripts that will need to be manually updated. This PR can't really be tested by using the builders, appart from the "not breaking do_build.sh" part.

Note2: This change is heavy, I don't expect this PR to be merged right away. In fact, we probably want to wait until after branching stable-6.

Feedback is welcome!

**Comment by jean-edouard** [comment]
Just added a commit to collect packages, see OXT-546.

File: build-scripts/setup.sh [hunk 1]
```diff
@@ -186,6 +208,20 @@ if [ ! -d "${BUILD_USER_HOME}"/ssh-key ]; then
...
 fi
 
+# Create build certs
+if [ ! -d "${BUILD_USER_HOME}"/certificates ]; then
+    mkdir "${BUILD_USER_HOME}"/certificates
+    openssl genrsa -out "${BUILD_USER_HOME}"/certificates/prod-cakey.pem 2048
+    openssl genrsa -out "${BUILD_USER_HOME}"/certificates/dev-cakey.pem 2048
+    openssl req -new -x509 -key "${BUILD_USER_HOME}"/certificates/prod-cakey.pem \
+            -out "${BUILD_USER_HOME}"/certificates/prod-cacert.pem -days 1095 \
+            -subj "/C=US/ST=Massachusetts/L=Boston/O=OpenXT/OU=OpenXT/CN=openxt.org"
```
Comment (pearsonk) [inline build-scripts/setup.sh]: Since OpenXT doesn't have a geological "home", probably should not use US/MA/Boston as part of the generic DN

**Comment by pearsonk** [comment]
setup.sh doesn't work for build systems behind a web proxy. Shouldn't necessarily stop this PR but will require changes for such an environment.

**Comment by rossphilipson** [comment]
This one has been ready to go for some time pending the stable-6 branching. That has happened so I will merge this later today if there are no more comments.

@jean-edouard can you create a ticket to address @pearsonk 's request for this to work with proxies?

File: build-scripts/setup.sh (hunk 1 above)
Comment (jean-edouard) [inline build-scripts/setup.sh]: Sorry, I guess I missed that comment.
I agree with you, but I'm pretty sure these values are mandatory, so I just set them to my current location...
What other values would you suggest?

**Pull request merged**